middle_end/flambda2/simplify/expr_builder.ml

```diff
@@ -58,16 +58,34 @@ let add_set_of_closures_offsets ~is_phantom named uacc =
       in
       UA.with_closure_offsets uacc (Known closure_offsets)
   in
+  let add_offsets_from_code uacc code =
+    match UA.closure_offsets uacc with
+    | Unknown -> uacc
+    | Known closure_offsets ->
+      let dacc = UA.creation_dacc uacc in
+      let code_id = Code.code_id code in
+      let from_function =
+        try Code_id.Map.find code_id (DA.closure_offsets dacc)
+        with Not_found ->
+          Misc.fatal_errorf "Missing offsets for code ID %a" Code_id.print
+            code_id
+      in
+      let closure_offsets =
+        Closure_offsets.add_offsets_from_function closure_offsets ~from_function
```

Comment:
In this case, we do not pass the `~is_phantom` information. I'm not sure whether we can have phantom code bindings, but either we need an assert that `not is_phantom` holds, or pass the `is_phantom` information to `Closure_offsets` so that phantom bindings can be ignored.

Reply:
We cannot have phantom code bindings. The `is_phantom` information is already present in the computed offsets (remember, we're not adding a constraint here, just aggregating already computed ones).